memegen/templates/index.html

```diff
@@ -140,87 +140,13 @@ <h3 class="masthead-brand">MemeGen.link</h3>
-
-    <script type="text/javascript">
-
-      function encodeMemeText(str) {
-          return str
-            .replace(/-/g, '--')
-            .replace(/_/g, '__')
-            .replace(/\?/g, '~q')
-            .replace(/%/g, '~p')
-            .replace(/\s+/g, '-') || '_';
-        }
-
-      function decodeMemeText(str) {
-          return str
-            .replace(/-/g, ' ')
-            .replace(/_/g, '')
-            .replace(/~q/g, '?')
-            .replace(/~p/g, '%')
-        }
-
-      function formatMeme(meme) {
-        if (!meme.element) { return meme.text; }
-        var source = encodeURI($(meme.element).attr('data-url')).replace(/,/g, '%2C').replace(/'/g, '%27');
-        var $meme = $(
-          "<div class='meme'><img height='50' src='" + source + "' /><span class='meme-text'>" + meme.text + "</span></div>"
-        );
-        return $meme;
-      };
-
-      function generateMeme() {
-        var memeId = $('.js-meme-selector').val();
-        var memeTextTop = encodeMemeText($('#meme-text-top').val());
-        var memeTextBottom = encodeMemeText($('#meme-text-bottom').val());
-        if (memeId && memeTextTop && memeTextBottom) {
-          var url = '/' + memeId + '/' + memeTextTop + '/' + memeTextBottom;
-          $("#meme-link").attr('href', url);
-          $("#meme-link").text(url);
-          $("#meme-image img").attr('src', encodeURI(url).replace(/,/g, '%2C').replace(/'/g, '%27') + '.jpg');
-          $("#meme-image").attr('href', encodeURI(url).replace(/,/g, '%2C').replace(/'/g, '%27') + '.jpg');
-        }
-      }
-
-      $('.js-meme-selector').select2({
-        templateResult: formatMeme,
-        templateSelection: formatMeme
-      });
-
-      $( ".js-meme-selector" ).on('change', function() {
-        var link = $( ".js-meme-selector option:checked").data('link');
-        if (link) {
-          var pieces = link.split('/');
-          $('#meme-text-top').val(decodeMemeText(pieces[2]));
-          $('#meme-text-bottom').val(decodeMemeText(pieces[3]));
-        }
-        generateMeme();
-      });
-
-      $("#btn-generate").on('click', function(e) {
-        generateMeme();
-      });
-
-      if ($('#meme-text-top').val() || $('#meme-text-bottom').val()) {
-        $(".js-meme-selector").trigger("change");
-      } else {
-        $(".js-meme-selector").val("{{ default_template }}").trigger("change");;
-      }
-    </script>
+    <script src="flask-api/static/js/default.js"></script>
 
     <script>
-      (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
-      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
-      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
-      })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
-      ga('create', '{{ ga_tid }}', 'auto');
-      ga('send', 'pageview');
+      $(".js-meme-selector").val("{{ default_template }}").trigger("change");
```

Comment:
In the old code, this was done conditionally so that if you navigate back to the homepage, your previous template selection and text remains.